lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -2402,25 +2453,30 @@ void ModuleEmitter::emitRTLModule(rtl::RTLModuleOp module) {
   SmallVector<rtl::RTLModulePortInfo, 8> portInfo;
   module.getRTLPortInfo(portInfo);
 
-  size_t nextPort = 0;
-  for (auto &port : portInfo)
-    addName(module.getArgument(nextPort++), port.name);
+  for (auto &port : portInfo) {
+    StringRef name = port.name.getValue();
+    if (name.empty()) {
+      llvm::errs() << "Found port without a name. Port names are required for "
+                      "Verilog synthesis.\n";
+      name = "<<NO-NAME-FOUND>>";
+    }
+    if (port.direction == rtl::PortDirection::OUTPUT)
+      usedNames.insert(name);
```

Comment:
I'm assuming (by context in other functions) that `usedNames.insert` copies strings into its data structure.

Reply:
Yep!